xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiTradeService.java

```diff
@@ -0,0 +1,177 @@
+    orderHistoryRequest.setStatus(0);
+    orderHistoryRequest.setPageNumber(1);
+    orderHistoryRequest.setPageSize(50);
+    try {
+      CoingiOrdersList list = getCoingiOrderHistory(orderHistoryRequest);
+      return CoingiAdapters.adaptOpenOrders(list);
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+  }
+
+  @Override
+  public String placeMarketOrder(MarketOrder order) throws IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
+
+  @Override
+  public String placeStopOrder(org.knowm.xchange.dto.trade.StopOrder stopOrder)
+      throws java.io.IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
+
+  @Override
+  public String placeLimitOrder(LimitOrder order) throws IOException {
+    CoingiPlaceLimitOrderRequest request =
+        new CoingiPlaceLimitOrderRequest()
+            .setCurrencyPair(CoingiAdapters.adaptCurrency(order.getCurrencyPair()))
+            .setOrderType(order.getType().equals(BID) ? 0 : 1)
+            .setPrice(order.getLimitPrice())
+            .setVolume(order.getRemainingAmount());
+
+    String response;
+    try {
+      response = placeCoingiLimitOrder(request).getResult();
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+
+    return response;
+  }
+
+  @Override
+  public boolean cancelOrder(String orderId) throws IOException {
+    // if it doesn't return an error and CoingiOrder is returned, then the cancellation is
+    // successful
+    CoingiCancelOrderRequest request = new CoingiCancelOrderRequest();
+    request.setOrderId(orderId);
+    CoingiOrder order;
+    try {
+      order = cancelCoingiOrder(request);
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+
+    return order != null;
+  }
+
+  @Override
+  public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
+    if (orderParams instanceof CancelOrderByIdParams) {
+      return cancelOrder(((CancelOrderByIdParams) orderParams).getOrderId());
+    } else {
+      return false;
+    }
+  }
+
+  /** Required parameter types: {@link TradeHistoryParamPaging#getPageLength()} */
+  @Override
+  public UserTrades getTradeHistory(TradeHistoryParams p) throws IOException {
+    CoingiOrdersList orderList;
+
+    if (p instanceof CoingiTradeHistoryParams) {
+      CoingiTradeHistoryParams params = (CoingiTradeHistoryParams) p;
+
+      CoingiGetOrderHistoryRequest request = new CoingiGetOrderHistoryRequest();
+
+      if (params.getCurrencyPair() != null) request.setCurrencyPair(params.getCurrencyPair());
+
+      request.setPageNumber(params.getPageNumber());
+      request.setPageSize(params.getPageSize());
+
+      if (params.getStatus() != null) request.setStatus(params.getStatus());
+
+      if (params.getType() != null) request.setType(params.getType());
+
+      try {
+        orderList = getCoingiOrderHistory(request);
```

Comment:
the try / catch CoingiException  hit not applied

Reply:
done, thanks for spotting this